ui/apps/platform/src/services/AdministrationEventsService.ts

```diff
@@ -0,0 +1,83 @@
+import axios from './instance';
+
+import { Pagination } from './types';
+
+const eventsCountUrl = '/v1/count/administration/events';
+const eventsUrl = '/v1/administration/events';
+
+export type AdministrationEvent = {
+    id: string;
+    type: AdministrationEventType;
+    level: AdministrationEventLevel;
+    message: string;
+    hint: string;
+    domain: string;
+    resourceType: string;
+    resourceId: string;
+    numOccurrences: string; // int64
+    lastOccurredAt: string; // ISO 8601
+    createdAt: string; // ISO 8601
+};
+
+export type AdministrationEventType =
+    | 'ADMINISTRATION_EVENT_TYPE_UNKNOWN'
+    | 'ADMINISTRATION_EVENT_TYPE_GENERIC'
+    | 'ADMINISTRATION_EVENT_TYPE_LOG_MESSAGE';
+
+export type AdministrationEventLevel =
+    | 'ADMINISTRATION_EVENT_LEVEL_UNKNOWN'
+    | 'ADMINISTRATION_EVENT_LEVEL_INFO'
+    | 'ADMINISTRATION_EVENT_LEVEL_SUCCESS'
+    | 'ADMINISTRATION_EVENT_LEVEL_WARNING'
+    | 'ADMINISTRATION_EVENT_LEVEL_ERROR';
+
+export type AdministrationEventsFilter = {
+    from: string; // ISO 8601 lower (older) boundary
+    until: string; // ISO 8601 upper (newer) boundary
+    domain: string;
+    resourceType: string;
+    type: AdministrationEventType;
+    level: AdministrationEventLevel;
+};
+
+export type CountAdministrationEventsRequest = {
+    filter: AdministrationEventsFilter;
+};
+
+// The total number of notifications after filtering and deduplication.
+export type CountAdministrationEventsResponse = {
+    count: string; // int64
+};
+
+export type GetAdministrationEventResponse = {
+    event: AdministrationEvent;
+};
+
+export type ListAdministrationEventsRequest = {
+    pagination: Pagination;
+    filter: AdministrationEventsFilter;
+};
+
+export type ListAdministrationEventsResponse = {
+    events: AdministrationEvent[];
+};
+
+// TODO CountAdministrationEventsRequest
+export function countAdministrationEvents(): Promise<string> {
+    return axios
+        .get<CountAdministrationEventsResponse>(eventsCountUrl)
+        .then((response) => response.data.count);
```

Comment:
I know we usually try to keep these service calls limited to just the request/response, but what do you think about parsing the number here and either:

1. Defaulting to `0` on a parse error
2. Rejecting the promise on a parse error and defaulting to `0` at the call site

Reply:
Dave's concern should also be passed to the folks working on this new API.

All the other count queries in our app have been returning numbers, for example,
https://stagingdb.demo.stackrox.com/main/apidocs#tag/DeploymentService/operation/DeploymentService_CountDeployments